File: 622.py
class MyCircularQueue:

    def __init__(self, k: int):
        self.q = [None] * k
        self.f_ptr = 0
        self.b_ptr = 0
        self.count = 0

    def enQueue(self, value: int) -> bool:
        if self.isFull(): 
            return False 
        else: 
            self.q[self.b_ptr] = value 
            self.b_ptr = (self.b_ptr + 1) % len(self.q)
            self.count += 1
        return True

    def deQueue(self) -> bool:
        if self.isEmpty(): 
            return False 
        else: 
            self.f_ptr = (self.f_ptr + 1) % len(self.q)
            self.count -= 1 
        return True

    def Front(self) -> int:
        if self.isEmpty(): 
            return -1 
        return self.q[self.f_ptr]

    def Rear(self) -> int:
        if self.isEmpty(): 
            return -1
        idx = (self.b_ptr + len(self.q) - 1) % len(self.q)
        return self.q[idx]

    def isEmpty(self) -> bool:
        return self.count == 0

    def isFull(self) -> bool:
        return self.count == len(self.q)


# Your MyCircularQueue object will be instantiated and called as such:
# obj = MyCircularQueue(k)
# param_1 = obj.enQueue(value)
# param_2 = obj.deQueue()
# param_3 = obj.Front()
# param_4 = obj.Rear()
# param_5 = obj.isEmpty()
# param_6 = obj.isFull()

File: 622.cpp
#include <algorithm>
#include <deque>
#include <iostream>
#include <map>
#include <set>
#include <unordered_map>
#include <unordered_set>
#include <vector>

using namespace std;

class MyCircularQueue {
public:

    vector<int> q;
    int f_ptr = 0; 
    int b_ptr = 0;
    uint64_t count = 0;

    MyCircularQueue(int k) {
        q.resize(k, -1); 
    }
    
    bool enQueue(int value) {
        if (isFull()) {
            return false;
        }
        q[b_ptr] = value;
        b_ptr = (b_ptr + 1) % q.size(); 
        count += 1; 
        return true; 
    }
    
    bool deQueue() {
        if (isEmpty()){
            return false; 
        }
        f_ptr = (f_ptr + 1) % q.size();
        count -= 1; 
        return true;
    }
    
    int Front() {
        if (isEmpty()) {
            return -1; 
        }
        return q[f_ptr];  
    }
    
    int Rear() {
        if (isEmpty()) {
            return -1; 
        }
        int idx = (b_ptr + q.size() - 1) % q.size(); 
        return q[idx];
    }
    
    bool isEmpty() {
        return count == 0; 
    }
    
    bool isFull() {
        return count == q.size(); 
    }
};

/**
 * Your MyCircularQueue object will be instantiated and called as such:
 * MyCircularQueue* obj = new MyCircularQueue(k);
 * bool param_1 = obj->enQueue(value);
 * bool param_2 = obj->deQueue();
 * int param_3 = obj->Front();
 * int param_4 = obj->Rear();
 * bool param_5 = obj->isEmpty();
 * bool param_6 = obj->isFull();
 */

int main() {}